src/julia-syntax.scm

```diff
@@ -1,6 +1,56 @@
 ;; ignored variable name. TODO replace with _?
 (define UNUSED '|#unused#|)
 
+;; simple stack implementation
+(define (make-stack)
+  (define stack '())
+
+  (define (empty-stack?)
+    (null? stack))
+
+  (define (push-stack! item)
+    (set! stack (cons item stack)))
+
+  (define (pop-stack!)
+    (if (empty-stack?)
+        (error "Stack is empty")
+        (let ((top (car stack)))
+          (set! stack (cdr stack))
+          top)))
+
+  (define (top-stack)
+    (if (empty-stack?)
+        '()
+        (car stack)))
+
+  (define (bottom-stack)
+    (if (empty-stack?)
+        '()
+        (let loop ((current stack))
+          (if (null? (cdr current))
+              (car current)
+              (loop (cdr current))))))
+
+  (define (display-stack)
+    (if (empty-stack?)
+        (display "Stack is empty")
+        (begin
+          (display "Stack contents: ")
+          (display stack)
+          (newline))))
+
+  (define (dispatch msg)
+    (case msg
+      ('push-stack! push-stack!)
+      ('pop-stack! pop-stack!)
+      ('empty-stack? empty-stack?)
+      ('top-stack top-stack)
+      ('bottom-stack bottom-stack)
+      ('display display-stack)
+      (else (error "Unknown operation" msg))))
+
+  dispatch)
```

Comment:
This stack machinery seems unnecessary compared to using the built-in data structures in scheme. See `parent-scope` in `macroexpand.scm`, which uses `cons` for push and `cdr`/`car` for pop

I don't really have a problem with it, but it's not very consistent with the existing style